src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedAutoPropAccessorSymbol.cs

```diff
@@ -0,0 +1,205 @@
+                        property.Location);
+
+                    var modifiers = ImmutableArray.Create(CSharpCustomModifier.CreateRequired(initOnlyType));
+                    type = type.WithModifiers(modifiers);
+                }
+                _voidType = new StrongBox<TypeWithAnnotations>(type);
+            }
+        }
+
+        public override TypeWithAnnotations ReturnTypeWithAnnotations
+        {
+            get
+            {
+                if (_voidType is object)
+                {
+                    Debug.Assert(MethodKind == MethodKind.PropertySet);
+                    return _voidType.Value;
+                }
+
+                return _property.TypeWithAnnotations;
+            }
+        }
+
+        public override MethodKind MethodKind => _accessorKind == AccessorKind.Get
+            ? MethodKind.PropertyGet
+            : MethodKind.PropertySet;
+
+        internal override bool IsInitOnly => _accessorKind == AccessorKind.Init;
+
+        public override int Arity => 0;
+
+        public override bool IsExtensionMethod => false;
+
+        public override bool HidesBaseMethodsByName => false;
+
+        public override bool IsVararg => false;
+
+        public override bool ReturnsVoid => _voidType is object;
+
+        public override bool IsAsync => false;
+
+        public override RefKind RefKind => RefKind.None;
+
+        public override FlowAnalysisAnnotations ReturnTypeFlowAnalysisAnnotations => FlowAnalysisAnnotations.None;
+
+        public override ImmutableArray<TypeWithAnnotations> TypeArgumentsWithAnnotations => ImmutableArray<TypeWithAnnotations>.Empty;
+
+        public override ImmutableArray<TypeParameterSymbol> TypeParameters => ImmutableArray<TypeParameterSymbol>.Empty;
+
+        public override ImmutableArray<ParameterSymbol> Parameters
+        {
+            get
+            {
+                if (_accessorKind == AccessorKind.Get)
+                {
+                    return ImmutableArray<ParameterSymbol>.Empty;
+                }
+                return ImmutableArray.Create(SynthesizedParameterSymbol.Create(
+                    this,
+                    _property.TypeWithAnnotations,
+                    ordinal: 0,
+                    RefKind.None,
+                    name: ParameterSymbol.ValueParameterName));
+            }
+        }
+
+        public override ImmutableArray<MethodSymbol> ExplicitInterfaceImplementations => ImmutableArray<MethodSymbol>.Empty;
+
+        public override ImmutableArray<CustomModifier> RefCustomModifiers => _property.RefCustomModifiers;
+
+        public override Symbol AssociatedSymbol => _property;
+
+        public override Symbol ContainingSymbol => _property.ContainingSymbol;
+
+        public override ImmutableArray<Location> Locations => _property.Locations;
+
+        public override Accessibility DeclaredAccessibility => _property.DeclaredAccessibility;
+
+        public override bool IsStatic => _property.IsStatic;
```

Comment:
I'd feel more comfortable with constant values for the various properties below that only return one value at the moment

Reply:
ping

---
In reply to: [441116988](https://github.com/dotnet/roslyn/pull/44883#discussion_r441116988) [](ancestors = 441116988)

Reply:
pong

---
In reply to: [441827331](https://github.com/dotnet/roslyn/pull/44883#discussion_r441827331) [](ancestors = 441827331,441116988)

Reply:
This symbol is supposed to be generalizable, in case we need to synthesize auto-properties in some other form later on. I prefer to let it dispatch to the property definition, since I think that's safer than hardcoding.